### PR TITLE
Premium Themes: Redirect to Thank You page when activating a Premium theme

### LINKED ISSUE #73835: Display Thank You page when activating a Premium theme
Parent task: 
- https://github.com/Automattic/wp-calypso/issues/72703

### PR DESCRIPTION
**Pull request by epeicher**
Resolves #73835

## Proposed Changes

- Redirect it to the Marketplace Thank You page when activating a Premium theme

## Testing Instructions

* Apply this branch in development which should have the feature flag `themes/display-thank-you-page` enabled
* Go to the themes page (`/themes/:site`)
* Select a premium theme and click on `Activate`
* You should be redirected to the Marketplace Thank You Page
* Disable the feature flag  (`/themes/:site?flags=-themes/display-thank-you-page `)
* Go to the themes page with the feature flag _disabled_: `/themes/:site?flags=-themes/display-thank-you-page `. Please note the `-` after the `flags=`
* Select a premium theme and click on `Activate`
* You should see the Thanks modal on the Themes page

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

File: client/my-sites/checkout/checkout-thank-you/marketplace/marketplace-thank-you-theme-section.tsx [hunk 1]
```diff
@@ -112,6 +112,13 @@ export const ThankYouThemeSection = ( { theme }: { theme: any } ) => {
 		[ siteId, theme ]
 	);
 
+	// Clear completed activated theme request state to avoid displaying the Thanks modal
+	useEffect( () => {
+		return () => {
+			dispatch( clearActivated( siteId ) );
+		};
+	}, [ dispatch, siteId ] );
```
Comment (WBerredo) [inline client/my-sites/checkout/checkout-thank-you/marketplace/marketplace-thank-you-theme-section.tsx]: On  #75546 I added it to `useThemesThankYouData` which I think should live the logic for anything theme related on Thank You page. What do you think?

**Review by WBerredo — COMMENTED** [review]
When trying the first time I got stuck into this step on the Thank You page, I couldn't reproduce it though.  For context, I was trying with a Premium site, dropping it here for future references.

CleanShot 2023-04-14 at 11 41 13@2x

File: client/my-sites/checkout/checkout-thank-you/marketplace/marketplace-thank-you-theme-section.tsx (hunk 1 above)
Comment (epeicher) [reply client/my-sites/checkout/checkout-thank-you/marketplace/marketplace-thank-you-theme-section.tsx]: I think `useThemesThankYouData` could be a better fit for this logic 👍 . So I will delete this piece from here and once https://github.com/Automattic/wp-calypso/pull/75546 is merged I can re-test

**Review by WBerredo — APPROVED** [review]
LGTM!

File: client/state/themes/actions/activate.js [hunk 2]
```diff
@@ -76,18 +76,24 @@ export function activate(
...
 		 */
-		const isPremiumTheme = isThemePremium( getState(), themeId );
 		const isExternallyManaged = isExternallyManagedTheme( getState(), themeId );
-		const isFreeTheme = ! isPremiumTheme && ! isExternallyManaged;
+		const isWooTheme = doesThemeBundleSoftwareSet( getState(), themeId );
 		const isDotComTheme = !! getTheme( getState(), 'wpcom', themeId );
-		if ( isFreeTheme && isDotComTheme && isEnabled( 'themes/display-thank-you-page' ) ) {
+		if (
+			isDotComTheme &&
+			isEnabled( 'themes/display-thank-you-page' ) &&
```
Comment (WBerredo) [inline client/state/themes/actions/activate.js:92]: Totally nitpicking: I would put the flag check as the first or last item of this `if`, just for the mental separation of this validation and the other *real* ones.

**Review by WBerredo — APPROVED** [review]
Its working as expected, LGTM!

File: client/state/themes/actions/activate.js (hunk 2 above)
Comment (epeicher) [reply client/state/themes/actions/activate.js:92]: Done

**Pull request merged**